README.md

```diff
@@ -0,0 +1,38 @@
+# Zirgen Circuit Compiler
+
+Zirgen is a compiler for a domain-specific language, also called "zirgen",
+which creates arithmetic circuits for the RISC Zero proof system.
+
+For the most part, users of RISC Zero should be writing their zkApps in Rust
+and using our RISC-V zkVM rather than building them directly in Zirgen.
+Sometimes, however, it is necessary or desirable to write parts of an
+application as an arithmetic circuit to integrate directly with our proof system
+and achieve better performance. In fact, we have an upcoming version of the zkVM
+circuit written in Zirgen! Zirgen will make it possible to build accelerators
+for important parts of your applications, or even to build entire other VMs that
+integrate into the RISC Zero ecosystem through proof composition. With that said,
+it's still a work in progress and has quite a few rough edges.
+
+[Getting Started](zirgen/docs/01_Getting_Started.md)
+
+[Language Overview](zirgen/docs/02_Conceptual_Overview.md)
+
+## RISC-V zkVM
+
+This repository also includes the source code of the
+[RISC-V circuit](zirgen/circuit/rv32im/v1/) included in our 1.0 release.
+
+## Circom integration
+
+We also have an integration with Circom in the works -- you'll be able to use
+the Zirgen compiler to accelerate verification of Circom witnesses for arbirary
```

Comment:
We've generally used "zirgen" to refer specifically to the compiler for the zirgen language.  The Circom stuff doesn't use this at all, though it does exist in this repositoy and shares parts of the compiler internals.  I don't know how much we care about the precision here.

Reply:
Seems fine to me for today, but I'd love to get the naming here dialed in before we make a zirgen-specific announcement.

In particular, I'd like to land on terminology that makes it easy for us to talk about this tooling in a user-centric way. 

i.e., it'd be great to be able to say something like "The _____ compiler enables users to build accelerator circuits for the RISC Zero zkVM in one of two languages: Circom or Zirgen." 

I chatted with Jacob about this topic and he suggested mb we can use the name `0STARK compiler` in order to indicate that it compiles to the `zkr` format which is associated with the `0STARK` proof system (which is the name we're giving to our STARK protocol)

Reply:
> "The _____ compiler enables users to build accelerator circuits for the RISC Zero zkVM using one of two languages: Circom or Zirgen."

I'm very agnostic about the right choices here, I just want to make sure that we have a way to talk about our circom integration in a way that feels clean.